Update doc to mention that XPath has direct connection to if comment aware checks are added or not

Identified at https://github.com/checkstyle/checkstyle/issues/6807#issuecomment-581674231 ,

XPath was generated for a single check and it suppressed all issues.

```
$ javac Test1.java
$ cat Test1.java 
public class Test1 {

    {}
    static {}

    void method6(int a) {
          {}; //ok
    }

    void method7(int a) {} //ok
    void method8(int a) {/*:)*/} //ok, should be violation

    class TestClass4 {} //ok

    enum Test {}

    interface Interface {}

    @interface ClassPreamble { } //ok, should be violation, as there is space
}

$ cat config.xml 
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE module PUBLIC "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
                        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">
        <module name="RightCurly">
             <property name="id" value="RightCurlyAlone"/>
             <property name="option" value="alone"/>
             <property name="tokens"
              value="CLASS_DEF, METHOD_DEF, CTOR_DEF, LITERAL_FOR, 
                     LITERAL_WHILE, STATIC_INIT,
                     INSTANCE_INIT, ANNOTATION_DEF, ENUM_DEF"/>
        </module>
        <module name="SuppressionXpathSingleFilter">
          <property name="id" value="RightCurlyAlone"/>
          <property name="query" value="//RCURLY[parent::SLIST[count(./*)=1]
                                                 or parent::OBJBLOCK[count(./*)=2]]"/>
        </module>
  </module>
</module>

$ java -jar /var/tmp/checkstyle-8.29-all.jar -c config.xml Test1.java 
Starting audit...
Audit done.
```

However, once a comment aware check is added, TodoComment, the suppression fails for one instance, which have comments in the violation path.

````
$ cat TestClass.java
public class Test1 {

    {}
    static {}

    void method6(int a) {
          {}; //ok
    }

    void method7(int a) {} //ok
    void method8(int a) {/*:)*/} //ok, should be violation

    class TestClass4 {} //ok

    enum Test {}

    interface Interface {}

    @interface ClassPreamble { } //ok, should be violation, as there is space
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">
        <module name="RightCurly">
             <property name="id" value="RightCurlyAlone"/>
             <property name="option" value="alone"/>
             <property name="tokens"
              value="CLASS_DEF, METHOD_DEF, CTOR_DEF, LITERAL_FOR, 
                     LITERAL_WHILE, STATIC_INIT,
                     INSTANCE_INIT, ANNOTATION_DEF, ENUM_DEF"/>
        </module>
        <module name="TodoComment" />
        <module name="SuppressionXpathSingleFilter">
          <property name="id" value="RightCurlyAlone"/>
          <property name="query" value="//RCURLY[parent::SLIST[count(./*)=1]
                                                 or parent::OBJBLOCK[count(./*)=2]]"/>
        </module>
  </module>
</module>

$ java -jar checkstyle-8.29-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[ERROR] TestClass.java:11:32: '}' at column 32 should be alone on a line. [RightCurlyAlone]
Audit done.
Checkstyle ends with 1 errors.
````

This shows a direct correlation between xpath suppressions and the type of checks added to a configuration. This is not very user friendly as users are directly aware if a check is comment aware or not. This could cause issues with generating a case that works for everyone.

--- a/src/site/xdoc/cmdline.xml.vm
+++ b/src/site/xdoc/cmdline.xml.vm
@@ -1009,6 +1009,15 @@ class Test {
   </p>
 </div>
 
+<p>
+  Note: The generated XPath query is specific to the configuration used during generation.
+  If you later add "comment-aware" checks (like <code>TodoComment</code>) to your
+  configuration, Checkstyle will insert comment nodes into the AST. This changes the
+  tree structure and may cause strict XPath queries (like those relying on child counts
+  or specific indices) to fail. It is recommended to regenerate the suppression XML
+  if you significantly change the active checks in your configuration.
+</p>
+
 <h4 id="Generating_Checks_And_Files_Suppressions_XML"><span>
 7. Generating Checks And Files Suppressions XML
 (<span class="no-transform">-G, --generate-checks-and-files-suppression</span>)

--- a/src/site/xdoc/filters/suppressionxpathfilter.xml
+++ b/src/site/xdoc/filters/suppressionxpathfilter.xml
@@ -209,6 +209,15 @@
           ATTENTION: filtering by message is dependent on runtime locale. If project is running
           in different languages it is better to avoid filtering by message.
         </p>
+
+        <p>
+          <b>Attention:</b> The structure of the Abstract Syntax Tree (AST) can change
+          depending on the enabled checks. If &quot;comment-aware&quot; checks (such as
+          <code>TodoComment</code>) are enabled, comment nodes are added to the AST.
+          XPath queries should be written robustly to handle the potential presence
+          of comment nodes. For example, relying on strict <code>count(*)</code>
+          checks might fail if comments are present in the code block.
+        </p>
       </subsection>
       <subsection name="Properties" id="SuppressionXpathFilter_Properties">
         <div class="wrapper">

--- a/src/site/xdoc/filters/suppressionxpathfilter.xml.template
+++ b/src/site/xdoc/filters/suppressionxpathfilter.xml.template
@@ -22,6 +22,15 @@
             <param name="modulePath"
                    value="src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionXpathFilter.java"/>
         </macro>
+
+        <p>
+          <b>Attention:</b> The structure of the Abstract Syntax Tree (AST) can change
+          depending on the enabled checks. If "comment-aware" checks (such as
+          <code>TodoComment</code>) are enabled, comment nodes are added to the AST.
+          XPath queries should be written robustly to handle the potential presence
+          of comment nodes. For example, relying on strict <code>count(*)</code>
+          checks might fail if comments are present in the code block.
+        </p>
       </subsection>
       <subsection name="Properties" id="SuppressionXpathFilter_Properties">
         <div class="wrapper">

